nameko/containers.py

```diff
@@ -171,8 +170,8 @@ def __init__(self, service_cls, config, worker_ctx_cls=None):
         self.started = False
         self._worker_pool = GreenPool(size=self.max_workers)
 
-        self._active_threads = {}
-        self._protected_threads = set()
+        self._worker_threads = WeakKeyDictionary()
```

Comment:
The `WorkerContext` has a reference to the `ServiceContainer`, and these dicts are on the container. I _think_ we're disciplined enough about the worker lifecycle to make these non-weak references, but I'm not sure.

I'm don't know what best practice is here -- should we make them weakrefs just in case, or is that sloppy?